ui/actions/nuxeo-move-content-to-coldstorage-button.js

```diff
@@ -0,0 +1,169 @@
+          </div>
+          <p class="horizontal">
+             [[i18n('documentContent.moveToColdStorage.description')]]
+          </p>
+          </div>
+
+          <div class="buttons">
+           <paper-button id="cancel" name="cancel" noink dialog-dismiss>
+             [[i18n('documentContent.moveToColdStorage.cancel')]]
+            </paper-button>
+           <paper-button id="confirm" name="confirm" class="primary" noink dialog-confirm on-tap="_toggle">
+              [[i18n('documentContent.moveToColdStorage.confirm')]]
+           </paper-button>
+          </div>
+        </nuxeo-dialog>
+
+         <nuxeo-dialog id="cannot_move_content_dialog" with-backdrop>
+          <p class="horizontal">
+             [[i18n('documentContent.moveToColdStorage.error.cannotPerformMove')]]
+          </p>
+          </div>
+
+          <div class="buttons">
+           <paper-button id="cancel" name="cancel" noink dialog-dismiss>
+             [[i18n('documentContent.moveToColdStorage.ok')]]
+           </paper-button>
+          </div>
+        </nuxeo-dialog>
+      `;
+    }
+
+    static get is() {
+      return 'nuxeo-move-content-to-coldstorage-button';
+    }
+
+    static get properties() {
+      return {
+        /**
+         * Input document.
+         */
+        document: Object,
+
+        /**
+         * Input user.
+         */
+        user: Object,
+      };
+    }
+
+    _toggleDialog() {
+      this.$.content_to_move_dialog.toggle();
+    }
+
+    _canMoveToColdStorage(document) {
+      return (
+        document &&
+        !this.hasFacet(document, 'ColdStorage') &&
+        (this.hasAdministrationPermissions(this.user) || this.hasPermission(document, 'WriteColdStorage')) &&
+        document.properties['file:content']
+      );
+    }
+
+    _toggle() {
+      if (this._canMoveToColdStorage(this.document)) {
+        this.$.opMove.execute().then(
+          (doc) => {
+            this.dispatchEvent(
+              new CustomEvent('document-updated', {
+                composed: true,
+                bubbles: true,
+                detail: { doc },
+              }),
+            );
+
+            this.dispatchEvent(
+              new CustomEvent('notify', {
+                composed: true,
+                bubbles: true,
+                detail: { message: [[this.i18n('documentContent.moveToColdStorage.success')]] },
+              }),
+            );
+          },
+          () => {
+            this.dispatchEvent(
+              new CustomEvent('notify', {
+                composed: true,
+                bubbles: true,
+                detail: { message: [[this.i18n('documentContent.moveToColdStorage.error')]] },
+              }),
+            );
+          },
+        );
+      } else {
+        this.$.cannot_move_content_dialog.toggle();
+      }
+    }
+  }
+
+  customElements.define(MoveToColdStorage.is, MoveToColdStorage);
+  Nuxeo.ExportButton = MoveToColdStorage;
```

Comment:
According to our newest guidelines, we don't want to expose elements on Nuxeo global namespace 😉